tests/backend_test.py

```diff
@@ -1234,3 +1236,41 @@ def test_ecr_gate_compilation(ibm_sherbrooke_backend: IBMQBackend) -> None:
             circ, optimisation_level
         )
         assert ibm_sherbrooke_backend.valid_circuit(compiled_circ)
+
+
+# helper function for testing
+def _get_qiskit_statevector(qc: QuantumCircuit) -> np.ndarray:
+    """Given a QuantumCircuit, use aer_simulator_statevector to compute its
+    statevector, return the vector with its endianness adjusted"""
+    back = Aer.get_backend("aer_simulator_statevector")
+    qc.save_state()
+    job = back.run(qc)
+    return np.array(job.result().data()["statevector"].reverse_qargs().data)
+
+
+# https://github.com/CQCL/pytket-qiskit/issues/99
+def test_statevector_simulator_gateset() -> None:
+    sv_backend = AerStateBackend()
+    sv_supported_gates = sv_backend.backend_info.gate_set
+    assert OpType.Reset and OpType.Measure in sv_supported_gates
+    assert OpType.Conditional in sv_supported_gates
+    circ = Circuit(3, 1)
+    circ.CCX(*range(3))
+    circ.U1(1 / 4, 2)
+    circ.H(2)
+    circ.Measure(2, 0)
+    circ.CZ(0, 1, condition_bits=[0], condition_value=1)
+    circ.add_gate(OpType.Reset, [2])
+    compiled_circ = sv_backend.get_compiled_circuit(circ)
+    assert sv_backend.valid_circuit(compiled_circ)
+    tket_statevector = sv_backend.run_circuit(compiled_circ).get_state()
+    qc = tk_to_qiskit(compiled_circ)
+    qiskit_statevector = _get_qiskit_statevector(qc)
+    assert compare_statevectors(tket_statevector, qiskit_statevector)
```

Comment:
Was this circuit intentionally constructed to have a deterministic statevector?
I'm not sure how to interpret the result in general, but what would happen if there were a layer of hadamards at the start, for example? Would the statevector be probabilistic?

Reply:
Yes this circuit is deterministic. Its taken from a paper and is part of a circuit that Lewis and I want to run for our IEEE demo.

Indeed if we inserted some Hadamards in front then the statevector test would sometimes fail as the circuit is non-deterministic.

I don't think this is a problem really as our `AerStateBackend` is a wrapper around qiskit's Aer statevector simulator. Given that they already support these non-unitary ops we are just ensuring that we support the same ops that qiskit can.

Reply:
Good idea. I'll do that. Thanks Alec.